### PR TITLE
feat(component): [statistic] newly added

### PR DESCRIPTION
**Pull request by webvs2**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by webvs2** [comment]
@chenxch How did it go this time?

**Comment by chenxch** [comment]
Good job, there are still some differences in the UI from the design draft, invalid comments in the code can be deleted, and then why the `prettierrc` needs to be modified, and finally the rendering can be displayed on the description, thanks for your hard work.

**Comment by chenxch** [comment]
After all these are processed, a code review will be conducted.

**Comment by webvs2** [comment]
Thank you for your guidance, ui will continue to improve😊

**Comment by webvs2** [comment]
`prettierrc`  Before, The default value was two digits, but when I restored the design, I found that this was not practical and changed it to null, I think users will set it when they need it, so I don't change it until then, do you think

**Comment by webvs2** [comment]
@chenxch   `prettierrc`  I have recovered the file. Eliminate useless code  and I made some changes to the ui.

**Comment by chenxch** [comment]
Maybe we can standardize the title of PR first, click edit on the right side of the title.
e.g.
```
feat(component): [statistic] newly added
```

File: docs/examples/statistic/card.vue [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    message: 'Are you going far away?',
+    type: 'success',
+  })
+}
+
+// code here
+</script>
+
+<style lang="scss" scoped>
+>>> .example-showcase {
```
Comment (chenxch) [inline docs/examples/statistic/card.vue:126]: Uniform use `:deep()`

File: docs/en-US/component/statistic.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+
+### Statistic Attributes
+
+| Attribute          | Description            | Type            | Accepted Values                 | Default   |
+|-------------  |---------------- |---------------- |---------------------- |-------- |
+| value            | Numerical content                 | string \| number    | -          | -       |
+| decimal-separator | Setting the decimal point            | string                  | -      | .       |
+| formatter        | Custom numerical presentation| v-slot \|({value}) => VNode | -   | -      |
```
Comment (chenxch) [inline docs/en-US/component/statistic.md:44]: should be `format`?

File: packages/components/statistic/__tests__/statistic.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { mount } from '@vue/test-utils'
```
Comment (chenxch) [inline packages/components/statistic/__tests__/statistic.spec.ts:1]: `jsx` should be used.

Comment (chenxch) [reply packages/components/statistic/__tests__/statistic.spec.ts:1]: I remember that the template generated should be a JSX file.

File: packages/components/statistic/src/statistic.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    type: [String, Number],
+    default: '',
+  },
+  timeIndices: {
+    type: Boolean,
+    default: false,
+  },
+  valueStyle: {
+    type: Object,
+    default() {
```
Comment (chenxch) [inline packages/components/statistic/src/statistic.ts:38]: default: () => ({})

File: docs/examples/statistic/card.vue (hunk 1 above)
Comment (webvs2) [reply docs/examples/statistic/card.vue]: eliminate

File: docs/en-US/component/statistic.md (hunk 2 above)
Comment (webvs2) [reply docs/en-US/component/statistic.md]: Yes, it has been modified

**Comment by webvs2** [comment]
@chenxch I think we can review it again

File: packages/components/statistic/src/index.vue [hunk 5]
```diff
@@ -0,0 +1,143 @@
+<template>
+  <div :class="[ns.b()]">
+    <div class="head">
+      <slot name="title">
+        <span class="title">
```
Comment (kooriookami) [inline packages/components/statistic/src/index.vue:5]: Class should have prefix, like `:class="ns.e('title')"`.

File: packages/components/statistic/src/index.vue [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+          {{ title }}
+        </span>
+      </slot>
+    </div>
+    <div class="con">
```
Comment (kooriookami) [inline packages/components/statistic/src/index.vue:10]: Please use a full word instead of abbr.

File: packages/components/statistic/src/index.vue [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+      <span class="prefix">
```
Comment (kooriookami) [inline packages/components/statistic/src/index.vue:11]: Need prefix here.

File: packages/components/statistic/src/index.vue [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        <slot name="prefix">
+          {{ prefix }}
+        </slot>
+      </span>
+
+      <span class="number" :style="valueStyle">
```
Comment (kooriookami) [inline packages/components/statistic/src/index.vue:17]: Same here.

File: packages/components/statistic/src/index.vue [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+        <slot name="formatter"> {{ disposeValue }}</slot>
+      </span>
+      <span class="suffix">
```
Comment (kooriookami) [inline packages/components/statistic/src/index.vue:20]: Same here.

File: packages/theme-chalk/src/statistic.scss [hunk 10]
```diff
@@ -0,0 +1,52 @@
+@use 'mixins/mixins' as *;
+@use 'mixins/var' as *;
+@use 'common/var' as *;
+
+@include b(statistic) {
+  @include set-component-css-var('statistic', $statistic);
+}
+
+@include b(statistic) {
```
Comment (kooriookami) [inline packages/theme-chalk/src/statistic.scss:9]: Styles like font-weight/font-style/color and so on, should use css var.

File: packages/theme-chalk/src/statistic.scss [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    align-items: center;
+    justify-content:center ;
+    color: #606266;
+  }
+
+  .con {
+    display: flex;
+    justify-content: center;
+    align-items: center ;
+  padding-top: 8px;
```
Comment (kooriookami) [inline packages/theme-chalk/src/statistic.scss:37]: Please format this code.

File: packages/theme-chalk/src/statistic.scss [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+  line-height: 1.5715;
+  list-style: none;
+  font-feature-settings: 'tnum';
+
+  .head {
+    padding: 0 5px;
+    font-style: normal;
+    font-weight: 400;
+    font-size: 12px;
+    letter-spacing: -0.01px;
```
Comment (kooriookami) [inline packages/theme-chalk/src/statistic.scss:25]: Why need `letter-spacing: -0.01px;`, What's the purpose?

**Comment by chenxch** [comment]
@webvs2 Do you have discord?

**Review by kooriookami — CHANGES_REQUESTED** [review]
I suggest to refer to `Empty` component, `Empty` is simple, but you can learn some standard writing styles from it.

File: packages/theme-chalk/src/statistic.scss (hunk 12 above)
Comment (webvs2) [reply packages/theme-chalk/src/statistic.scss:25]: I found this in the css in the draft and added it

Comment (kooriookami) [reply packages/theme-chalk/src/statistic.scss:25]: You could modify and remove any code just make sense.

Comment (webvs2) [reply packages/theme-chalk/src/statistic.scss:25]: OK, I'll adjust it👌

File: packages/components/statistic/src/index.vue (hunk 5 above)
Comment (webvs2) [reply packages/components/statistic/src/index.vue:5]: Thank you, I will use css in js style to adjust

**Comment by webvs2** [comment]
@chenxch I'm sorry, but for external reasons, I'm here at the moment 😭

File: packages/theme-chalk/src/statistic.scss [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    display: flex;
+    justify-content: center;
+    align-items: center ;
+  padding-top: 8px;
+    .number {
+      font-style: normal;
+      padding: 0 5px;
+      font-family: sans-serif;
+    }
+    span {
```
Comment (ryuhangyeong) [inline packages/theme-chalk/src/statistic.scss:43]: ## Suggest

A more specific class name is required, and refactoring using bem mxins is recommended.

example

```
// steps.scss
@use 'mixins/mixins' as *;
@use 'common/var' as *;

@include b(steps) {
  display: flex;

  @include m(simple) {
    padding: 13px 8%;
    border-radius: 4px;
    background: getCssVar('fill-color', 'light');
  }

  @include m(horizontal) {
    white-space: nowrap;
  }

  @include m(vertical) {
    height: 100%;
    flex-flow: column;
  }
}
```

File: docs/en-US/component/statistic.md [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+
+## Card usage
+
+
+:::demo Card usage display, can be freely combined
+
+statistic/card
+
+:::
+
```
Comment (ryuhangyeong) [inline docs/en-US/component/statistic.md:34]: ## Suggest

It seems necessary to use prettier to clean up unnecessary whitespace or code.

Comment (webvs2) [reply docs/en-US/component/statistic.md:34]: Thank you. I'll adjust it

File: packages/components/statistic/src/index.vue (hunk 7 above)
Comment (webvs2) [reply packages/components/statistic/src/index.vue]: `:class="ns.e('prefix')" ` Is that it? Or is it something else?

File: packages/components/statistic/src/index.vue [hunk 15]
```diff
@@ -0,0 +1,143 @@
+<template>
+  <div :class="[ns.b()]">
+    <div :class="ns.e('head')">
+      <slot name="title">
+        <span :class="ns.e('title')">
```
Comment (kooriookami) [inline packages/components/statistic/src/index.vue:5]: It seems written backwards.
```js
<span :class="ns.e('title')">
  <slot name="title">
```

**Review by kooriookami — CHANGES_REQUESTED** [review]

**Comment by kooriookami** [comment]
Unit Test is failed, please take a look. :D

File: packages/components/statistic/src/index.vue (hunk 15 above)
Comment (webvs2) [reply packages/components/statistic/src/index.vue:5]: At that time, the layout was considered to be controlled by the head, and the internal styles would not be preserved if they were customized

Comment (webvs2) [reply packages/components/statistic/src/index.vue:5]: At that time, the layout was considered to be controlled by the head, and the internal styles would not be preserved if they were customized

Comment (webvs2) [reply packages/components/statistic/src/index.vue:5]: What do you think

Comment (kooriookami) [reply packages/components/statistic/src/index.vue:5]: I consider `ns.e('title')` is a default class. Users can modify the styles or define something new to cover it. So do Other components like `Input` .

**Comment by webvs2** [comment]
@kooriookami  Fixed the issue. test and lint are already passing locally

**Comment by kooriookami** [comment]
image
Oops, Here is a bug.

**Comment by webvs2** [comment]
What's going on with this bug, not seeing any errors in the local and online playgrounds😥😥😥😭

File: docs/en-US/component/statistic.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+| time-indices | Whether to enable the countdown function | boolean | true\|false | false      |
+| value        | Required value, enter the bound value    | string  | —           | —          |
+| format       | Formatting the countdown display         | string  | —           | 'HH:mm:ss' |
+
+### Statistic.Countdown Events
+
+| Method | Description                                | Parameters       |
+| ------ | ------------------------------------------ | ---------------- |
+| change | Enable in the 'countdown' function         | (value: Date)    |
+| finish | Launched after the 'countdown' is complete | (value: boolean) |
```
Comment (ryuhangyeong) [inline docs/en-US/component/statistic.md:71]: ## Suggest

※ this is a future consideration, so you can do it later.

We are working on improving the documentation a bit better. Please refer to the link below and update!

https://github.com/element-plus/element-plus/pull/10680/files

File: packages/components/statistic/__tests__/statistic.spec.tsx [hunk 17]
```diff
@@ -0,0 +1,57 @@
+import { mount } from '@vue/test-utils'
```
Comment (ryuhangyeong) [inline packages/components/statistic/__tests__/statistic.spec.tsx:1]: ## Suggest

change filename (`statistic.spec.tsx` -> `statistic.test.tsx`)

File: docs/en-US/component/statistic.md (hunk 16 above)
Comment (webvs2) [reply docs/en-US/component/statistic.md:71]: The finish and change capabilities are included; now do you want to remove them?🧐

Comment (webvs2) [reply docs/en-US/component/statistic.md:71]: Or do you mean I should change the instructions in the future, Only about statistic.md

File: docs/en-US/component/statistic.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
+---
+title: statistic
+lang: en-US
+---
+
+# statistic
```
Comment (ryuhangyeong) [inline docs/en-US/component/statistic.md:6]: ## Suggest

`statistic` -> `Statistic`

File: packages/components/statistic/__tests__/statistic.spec.tsx (hunk 17 above)
Comment (webvs2) [reply packages/components/statistic/__tests__/statistic.spec.tsx:1]: ok

File: docs/en-US/component/statistic.md (hunk 16 above)
Comment (ryuhangyeong) [reply docs/en-US/component/statistic.md:71]: I'm sorry for the confusion.

## Example

```ts

// Add a comment as if it were in a link.
const props = defineProps({
  /**
   * @description layout direction for child elements
   */
  direction: {
    type: String,
  },
})
```

File: docs/en-US/component/statistic.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+| Method | Description                                | Parameters       |
+| ------ | ------------------------------------------ | ---------------- |
+| change | Enable in the 'countdown' function         | (value: Date)    |
+| finish | Launched after the 'countdown' is complete | (value: boolean) |
+
+### Statistic Methods
+
+| Method  | Description         | Parameters      | CallBack      |
+| ------- | ------------------- | --------------- | ------------- |
+| suspend | Pause the countdown | (value:boolean) | (value: Date) |
```
Comment (ryuhangyeong) [inline docs/en-US/component/statistic.md:77]: ## Suggest

`(value:boolean)` -> `(value: boolean)`

File: docs/en-US/component/statistic.md (hunk 16 above)
Comment (webvs2) [reply docs/en-US/component/statistic.md]: Are you better now？

File: packages/components/statistic/src/index.vue [hunk 20]
```diff
@@ -0,0 +1,143 @@
+<template>
+  <div :class="[ns.b()]">
+    <div :class="ns.e('head')">
+      <span :class="ns.e('title')">
```
[the remark this answers is not in the document]
Comment (webvs2) [reply packages/components/statistic/src/index.vue:4]: Thanks, ok, I'll adjust

**Comment by kooriookami** [comment]
image
If the content is empty, `el-statistic__content` should be set to `v-if=false`. Also, if no prefix, hide `el-statistic__prefix`; no number, hide `el-statistic__number`; no suffix, hide `el-statistic__suffix`. So do other values or slots.

File: docs/en-US/component/statistic.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+| value             | Numerical content              | ^[string] / ^[number]    | -               | -       |
+| decimal-separator | Setting the decimal point      | ^[string]                | -               | .       |
+| format            | Custom numerical presentation  | ^[string]                | -               | -       |
+| group-separator   | Sets the thousandth identifier | ^[string]                | -               | ,       |
+| precision         | numerical precision            | ^[number]                | -               | 0       |
+| prefix            | Sets the prefix of a number    | ^[string] / ^[Component] | -               | -       |
+| suffix            | Sets the suffix of a number    | ^[string] / ^[Component] | -               | -       |
+| title             | Numeric titles                 | ^[string] / ^[Component] | -               | -       |
+| value-style       | Styles numeric values          | ^[styleObject]           | -               | -       |
+| rate              | Set the ratio                  | ^[number]                | -               | 1000    |
```
[the remark this answers is not in the document]
Comment (webvs2) [reply docs/en-US/component/statistic.md:47]: I don't understand this. Can you give me an example

[the remark this answers is not in the document]
Comment (chenxch) [reply docs/en-US/component/statistic.md:47]: You can refer to the `input.md`

File: packages/components/statistic/src/statistic-custom.ts [hunk 22]
```diff
@@ -0,0 +1,11 @@
+import { computed } from 'vue'
+import { useNamespace } from '@element-plus/hooks'
+export function useStatisticCustomStyle() {
```
Comment (ryuhangyeong) [inline packages/components/statistic/src/statistic-custom.ts:3]: ## Suggest

this file doesn't appear to be used anywhere. how about deleting it?

Comment (webvs2) [reply packages/components/statistic/src/statistic-custom.ts:3]: OK, sure

**Comment by chenxch** [comment]
@webvs2 I have sent you the discord for the planning and design of the components, as well as the code refactoring patch, you can take a look at it tomorrow, and feel free to contact me if you have any questions.

File: packages/components/statistic/src/countdown.vue [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+import { useNamespace } from '@element-plus/hooks'
+import { isNumber } from '@element-plus/utils'
+import { countdownEmits, countdownProps, formatTimeStr } from './countdown'
+
+const REFRESH_INTERVAL = 1000 / 30
+
+defineOptions({
+  name: 'ElCountdown',
+})
+onBeforeUnmount(() => {
```
Comment (chenxch) [inline packages/components/statistic/src/countdown.vue:39]: Lifecycle functions can be moved before defineExpose

File: packages/components/statistic/src/statistic.ts [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  title: {
+    type: [String, Number],
+    default: '',
+  },
+  valueStyle: {
+    type: definePropType<StyleValue>([String, Object]),
+    default: () => ({}),
+  },
+  rate: {
```
Comment (chenxch) [inline packages/components/statistic/src/statistic.ts:40]: Is it better to use `groupStep` ?  /cc @kooriookami

File: packages/components/statistic/src/statistic.ts [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+    type: Number,
+    default: 3,
+  },
+} as const)
+export const regroup = function (
+  target: any,
+  _mulriple = 3,
+  _groupSeparator = ','
+): string {
```
Comment (chenxch) [inline packages/components/statistic/src/statistic.ts:49]: There is no need to start the name with an underscore here, why is the target type `any`?

File: packages/components/statistic/src/countdown.vue [hunk 26]
```diff
@@ -0,0 +1,91 @@
...
+watch(
+  () => [props.value, props.format],
+  () => {
+    stopTimer()
+    startTimer()
+  },
+  {
+    immediate: true,
+  }
+)
```
Comment (chenxch) [inline packages/components/statistic/src/countdown.vue:86]: You need to add a destroy event to destroy the timer, otherwise it will cause memory leaks.

File: packages/components/statistic/src/statistic.ts (hunk 24 above)
Comment (kooriookami) [reply packages/components/statistic/src/statistic.ts:40]: > Is it better to use `groupStep` ? /cc @kooriookami

Maybe `separatorCount` is more better.

Comment (kooriookami) [reply packages/components/statistic/src/statistic.ts:40]: Different countries use different way to separate numbers. The `rate` prop can not satisfy all cases. We should abandon it.

**Review by kooriookami — APPROVED** [review]
Migrate.

**Review by chenxch — APPROVED** [review]
